lua/chainsaw/core/log-commands.lua

```diff
@@ -104,5 +104,38 @@ function M.removeLogs()
 	vim.b.timelogStart = nil
 end
 
+function M.removeLogsVisual()
+	local marker = require("chainsaw.config.config").config.marker
+	local numOfLinesBefore = vim.api.nvim_buf_line_count(0)
+
+	-- Get the start and end of current visual selection
+	local selectedStart = vim.fn.getpos("v")
+	local selectedEnd = vim.fn.getpos(".")
+	if selectedStart[2] > selectedEnd[2] then
+		selectedStart, selectedEnd = selectedEnd, selectedStart
+	end
+
+	-- Remove lines only for visual selection
+	local bufLines = vim.api.nvim_buf_get_lines(0, selectedStart[2] - 1, selectedEnd[2], false)
+	for i = #bufLines, 1, -1 do
```

Comment:
How about running this loop from selectedEnd to selectedEnd? That way the iterator is automatically already the line number.